ESP32   get error "undefined reference to `sprintf' "  while CONFIG_NEWLIB_LIBC=y

**Describe the bug**
Get a undeclared errors "undefined reference to `sprintf' " after enabling CONFIG_NEWLIB_LIBC.
Get a undeclared error "'CONFIG_ESP32_PHY_MAX_TX_POWER' undeclared" while enable CONFIG_BT.

**Logs and console output**
zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp_wifi/lib/esp32/libphy.a(phy_chip_v7.o): in function `fe_reg_init':
/home/cff/gittree/chip7.1_phy/chip_7.1/board_code/app_test/pp/phy/phy_chip_v7.c:3382: undefined reference to `sprintf'

/home/sam/.local/zephyr-sdk-0.14.1/xtensa-espressif_esp32_zephyr-elf/bin/../lib/gcc/xtensa-espressif_esp32_zephyr-elf/10.3.0/../../../../xtensa-espressif_esp32_zephyr-elf/bin/ld.bfd: /home/sam/zephyrproject/modules/hal/espressif/zephyr/esp32/../../components/esp_wifi/lib/esp32/libphy.a(phy_chip_v7.o): in function `get_phy_version_str':
/home/cff/gittree/chip7.1_phy/chip_7.1/board_code/app_test/pp/phy/phy_chip_v7.c:3382: undefined reference to `sprintf'

@sylvioalves You can test the 'hello_world' Project with under config, you will get error "undefined reference to `sprintf'",  and , if you comment the flash config, will get error "undefined symbol `printf'".

CONFIG_BT=y
CONFIG_BT_DEVICE_NAME="zephyr esp32"
CONFIG_BT_DEVICE_APPEARANCE=384

# FLASH
CONFIG_FLASH=y
CONFIG_FLASH_MAP=y
CONFIG_FLASH_PAGE_LAYOUT=y

CONFIG_NEWLIB_LIBC=y

--- a/west.yml
+++ b/west.yml
@@ -65,7 +65,7 @@ manifest:
       groups:
         - hal
     - name: hal_espressif
-      revision: bceaadfc8f4f812ace778937a6b042e3b6249215
+      revision: df85671c5d0405c0747c2939c8dfe808b7e4cf38
       path: modules/hal/espressif
       west-commands: west/west-commands.yml
       groups:

